channels/utils.py

```diff
@@ -289,3 +289,38 @@ def num_items_not_none(items):
     """
 
     return len(list(filter(lambda val: val is not None, items)))
+
+
+def _render_article_nodes(node):
+    """
+    Render article nodes into pieces of text
+
+    Args:
+        node (list or dict): A node in the article content
+
+    Yields:
+        str: A string containing text from the article nodes
+    """
+    if isinstance(node, list):
+        for item in node:
+            yield from _render_article_nodes(item)
+    elif isinstance(node, dict):
+        if "text" in node:
+            yield node["text"]
+        if "children" in node:
+            yield from _render_article_nodes(node["children"])
+        if node.get("name") == "p":
+            yield " "
+
+
+def render_article_text(content):
+    """
+    Render article text based on the article content data structure
+
+    Args:
+        content (list or dict): The article content
+
+    Returns:
+         str: A string containing the text from the text nodes of the article
+    """
+    return "".join(_render_article_nodes(content))
```

Comment:
Maybe change to `" ".join`?  Otherwise some words get merged together.  For example, this article:
<img width="623" alt="screen shot 2019-01-02 at 2 57 14 pm" src="https://user-images.githubusercontent.com/187676/50610443-018c3480-0ea1-11e9-99cf-bae1616a499f.png">

leads to this in the index:
`"article_text": "Crazy1 crazy4crazy5crazy6 crazy7 crazy2crazy3"`

Reply:
I changed it to add spaces around lists and tables. I think we can't do `" ".join` since sometimes there are two pieces of text which are right next to each other which need to be concatenated